### PR TITLE
Fetching messages

### PR DESCRIPTION
**Pull request by monikasudol**

### CODE REVIEW

File: src/state/users/index.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  username: 'Asia'
+}];
+
+export const FETCH_USERS = 'users: fetch-users';
+const FETCH_USERS_SUCCESS = 'users: fetch-users-success';
+const FETCH_USERS_FAILURE = 'users: fetch-users-failure';
+
+export const usersReducer = handleActions({
+
+  [FETCH_USERS_SUCCESS]: (state, payload) => {[
```
[the remark this answers is not in the document]
Comment (monikasudol) [reply src/state/users/index.js]: done

[the remark this answers is not in the document]
Comment (monikasudol) [reply src/state/users/index.js]: done

File: src/api/messages.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { toJSON } from '../utils/index';
+
+export const fetchMessages = ({ id, timestamp }) => fetch(
+  'http://localhost:8080' + createGetMessageQueryString(id, timestamp)
+).then(toJSON);
+
+const createGetMessageQueryString = (id, timestamp) => {
```
[the remark this answers is not in the document]
Comment (monikasudol) [reply src/api/messages.js]: resolved

File: src/api/messages.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { toJSON } from '../utils/index';
+
+export const fetchMessages = ({ id, timestamp }) => fetch(
```
[the remark this answers is not in the document]
Comment (monikasudol) [reply src/api/messages.js]: done

File: src/modules/main/message-field/messages/message/index.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import React, { Component } from 'react';
+import './style.css';
+
+class Message extends Component {
```
[the remark this answers is not in the document]
Comment (monikasudol) [reply src/modules/main/message-field/messages/message/index.js]: improved

File: src/modules/main/navigator/channels/index.js [hunk 5]
```diff
@@ -16,7 +16,7 @@ class Channels extends Component {
 
   handleAddingChannel = () => {
     this.setState({
-      isChannelCreated: true
+      beingCreated: true
```
[the remark this answers is not in the document]
Comment (monikasudol) [reply src/modules/main/navigator/channels/index.js]: improved

File: src/state/messages/index.js [hunk 6]
```diff
@@ -1,18 +1,28 @@
...
+const CLEAR_MESSAGE_BOARD = 'messages: clear-message-board';
 export const FETCH_MESSAGES = 'messages: fetch-messages';
 export const FETCH_MESSAGES_SUCCESS = 'messages: fetch-messages-success';
 const FETCH_MESSAGES_FAILURE = 'messages: fetch-messages-failure';
 
 export const messagesReducer = handleActions({
 
+  [CLEAR_MESSAGE_BOARD]: (state, { payload }) => ({
+    ...state,
+    messages: [{}]
```
Comment (yuriyyakym) [inline src/state/messages/index.js]: Do we need this empty object here? Looks strange.

File: src/state/messages/saga.js [hunk 7]
```diff
@@ -1,33 +1,31 @@
...
-
-  }
-};
 
-function* onFetchingNewMessages(action) {
-  const { payload } = action;
   while (true) {
-    const messages = yield call(messagesApi.fetchMessages, payload);
-    yield put(fetchMessagesSuccess(messages));
     yield delay(1000);
```
Comment (yuriyyakym) [inline src/state/messages/saga.js]: Do we need this delay?

File: src/state/messages/selectors.js [hunk 8]
```diff
@@ -1 +1,34 @@
...
+);
+
+export const selectConversationist = state => state.messages.conversetionist;
+
+const findUsername = (userId, users) => {
+  const messageSender = users.find(user => user.id === userId);
+  return messageSender.username;
+};
+
+export const addUsernamesToMessage = (state, messages) => {
```
Comment (yuriyyakym) [inline src/state/messages/selectors.js]: Maybe something like:
```
const selectDetailedMessages = createSelector(
  [ selectUsers, selectMessages ],
  (users, messages) => messages.map(message => ({
    ...message,
    username: users.find(user => user.id === message.userId)
  }));
);
```

What do you think?

**Comment by michal-marciniec** [comment]
Post-midnight commits ❤️😄

File: package.json [hunk 9]
```diff
@@ -11,9 +11,11 @@
     "react-router": "^4.2.0",
     "react-router-dom": "^4.2.2",
     "react-scripts": "1.1.4",
+    "react-scrollbar": "^0.5.4",
     "redux": "^4.0.0",
     "redux-actions": "^2.4.0",
     "redux-saga": "^0.16.0",
+    "redux-select": "^3.5.2",
```
[the remark this answers is not in the document]
Comment (monikasudol) [reply package.json]: actually not used, i've removed

File: src/api/messages.js [hunk 10]
```diff
@@ -1,9 +1,20 @@
...
+// get(
+//   'http://localhost:8080' + createGetMessageQueryString(id, timestamp)
+// ).then(toJSON);
 
 const createGetMessageQueryString = (id, timestamp) => {
-  `/messages/private/${id}?newerThan=${timestamp}`
+  if (timestamp) {
+    return `/messages/private/${id}?newerThan=${timestamp}`;
+  }
+  else {
```
[the remark this answers is not in the document]
Comment (monikasudol) [reply src/api/messages.js]: how could it happen? :D

File: src/modules/main/message-field/messages/index.js [hunk 11]
```diff
@@ -1,30 +1,37 @@
...
     return (
       <div className='messages-container'>
-        {messages.length > 0 && (
-          <div>
-            {messages.map((message, index) => (
-              <Message text={message}
-                key={index} />
-            ))}
-          </div>)}
+        {(messages.length > 0) && (
```
[the remark this answers is not in the document]
Comment (monikasudol) [reply src/modules/main/message-field/messages/index.js]: Right, improved

File: src/state/messages/index.js (hunk 6 above)
Comment (monikasudol) [reply src/state/messages/index.js]: You're rught, but I don't know why, but it doesn't work without it.

File: src/state/messages/saga.js (hunk 7 above)
Comment (monikasudol) [reply src/state/messages/saga.js]: Yes, doesn't work without, but I've removed second delay from `try.. catch..`

File: src/state/messages/selectors.js (hunk 8 above)
Comment (monikasudol) [reply src/state/messages/selectors.js]: Yeah, better, but unfortunately not applicable in this case or I do sth wrong.. I have left it for now, but i will be think about it

File: src/components/input-wrapper/index.js [hunk 12]
```diff
@@ -1,5 +1,5 @@
 import React, { Component } from 'react';
-import classNames from 'classnames';
+import classnames from 'classnames';
```
Comment (yuriyyakym) [inline src/components/input-wrapper/index.js]: `classNames` was a better name :)

File: src/modules/main/message-field/messages/message/index.js [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+  return (
+    <div className='message-container' >
+      <span className='message-conversationist-name'>
+        {username}
+      </span>
+      <span className='message-text'>
+        {text}
+      </span>
+    </div >
```
Comment (yuriyyakym) [inline src/modules/main/message-field/messages/message/index.js]: redundant space

File: src/modules/main/message-field/messages/message/index.js [hunk 14]
```diff
@@ -0,0 +1,18 @@
+import React, { Component } from 'react';
+import './style.css';
+
+const Message = ({ text, username }) => {
+
+  return (
```
Comment (yuriyyakym) [inline src/modules/main/message-field/messages/message/index.js]: I prefer arrow functions without return `statement`. But it's up to you :)

**Pull request merged**